src/nvim/screen.c

```diff
@@ -5278,7 +5278,7 @@ void screen_puts_len(char_u *text, int textlen, int row, int col, int attr)
   int force_redraw_next = FALSE;
   int need_redraw;
 
-  const int l_has_mbyte = has_mbyte;
+  const bool l_has_mbyte = has_mbyte;
```

Comment:
There are are lot's of references to `has_mbyte`, 388 in total on this branch, and it would not be reasonable to remove them all in one PR. `screen.c` is surely a module that could be simplified under the `enc=utf-8` assumption, but I rather do it in a separate PR where it will get sufficient attention by its own, after this one is merged. (though the addition of a lot more screen tests should make us a bit more confident of this sort of changes than when this PR was first proposed)
This PR needs review for the existing changes, rather than weighing it down with further changes spread across the codebase.